### PR TITLE
Add form

### PR DESCRIPTION
**Pull request by BiassedTomato**
## Summary

Added search form to the main page 

### Screenshot

image

## How this was tested?

Manual testing

### List of steps to manually test introduced functionality:

-   Go to http://localhost:3000
-   Enter ingredients and calories
-   Press enter/click "Submit"

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have added tests for new functionality
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generate no new warnings
-   [x] My pull-request is shorter than 500 lines

### CODE REVIEW

**Review by VladiShcherbakov — APPROVED** [review]

File: src/components/MainComponent/MainComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import MainDiv from "./styles";
+import fetchRecipes from "../../api/fetchRecipes";
+
+const MainComponent = () => {
+  const search = async (ingredient = "", caloriesLimit = 0) => {
+    const trimmedIngredients = ingredient.replaceAll(",", "");
+
+    if (ingredient.length < 1 || caloriesLimit < 1) return [];
+
+    const result = await fetchRecipes(trimmedIngredients, caloriesLimit);
```
Comment (nadezhdakharchuk) [inline src/components/MainComponent/MainComponent.jsx:13]: Could you please use `try/catch` construction here and catch error

File: src/components/RecipesList/RecipesList.jsx [hunk 2]
```diff
@@ -4,41 +4,45 @@ import fetchRecipes from "../../api/fetchRecipes";
 
 import List from "./styles";
 
-const RecipesList = () => {
-  const [recipes, setRecipes] = useState([]);
+const RecipesList = (props) => {
+  const [cards, setCards] = useState([]);
   const [error, setError] = useState(null);
 
+  const { recipeList } = props;
```
Comment (nadezhdakharchuk) [inline src/components/RecipesList/RecipesList.jsx:11]: `const RecipesList = ({ recipeList }) =>`

File: src/components/RecipesList/RecipesList.jsx [hunk 3]
```diff
@@ -4,41 +4,45 @@ import fetchRecipes from "../../api/fetchRecipes";
...
+
+  const createCards = async (cardRecipeList = []) => {
+    const recipes =
+      cardRecipeList.length > 0 ? cardRecipeList : await fetchRecipes();
+
+    return recipes.map(({ recipe }) => (
+      <RecipeItem
```
Comment (nadezhdakharchuk) [inline src/components/RecipesList/RecipesList.jsx:18]: Why do we use render items here?

**Review by nadezhdakharchuk — CHANGES_REQUESTED** [review]

File: src/components/MainComponent/MainComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from "react";
+import RecipesList from "../RecipesList";
+import SearchBox from "../SearchBox";
+import MainDiv from "./styles";
+import fetchRecipes from "../../api/fetchRecipes";
+
+const MainComponent = () => {
+  const search = async (ingredient = "", caloriesLimit = 0) => {
```
Comment (VladimirAfanasievFS) [inline src/components/MainComponent/MainComponent.jsx:8]: you can extract this function outside module

**Pull request merged**